sdks/python/apache_beam/runners/portability/portable_runner.py

```diff
@@ -95,65 +96,28 @@ def default_docker_image():
   @staticmethod
   def _create_environment(options):
     portable_options = options.view_as(PortableOptions)
-    environment_urn = common_urns.environments.DOCKER.urn
-    if portable_options.environment_type == 'DOCKER':
-      environment_urn = common_urns.environments.DOCKER.urn
-    elif portable_options.environment_type == 'PROCESS':
-      environment_urn = common_urns.environments.PROCESS.urn
-    elif portable_options.environment_type in ('EXTERNAL', 'LOOPBACK'):
-      environment_urn = common_urns.environments.EXTERNAL.urn
-    elif portable_options.environment_type:
-      if portable_options.environment_type.startswith('beam:env:'):
-        environment_urn = portable_options.environment_type
-      else:
+    if not portable_options.environment_type:
+      environment_type = 'DOCKER'
+    else:
+      environment_type = portable_options.environment_type
+
+    if environment_type.startswith('beam:env:'):
+      environment_urn = environment_type
+    else:
+      try:
+        environment_urn = environments.Environment.get_urn_from_type(
+          environment_type)
```

Comment:
I'd like to get rid of this lookup.  Instead, I'd like to do something like:

```python
environment_type = portable_options.environment_type
if not environment_type:
  environment_urn = common_urns.environments.DOCKER.urn
elif environment_type.startswith('beam:env:'):
  environment_urn = environment_type
else:
  # e.g. handle LOOPBACK -> EXTERNAL
  environment_type = ENV_TYPE_ALIASES.get(environment_type, environment_type)
  environment_urn = 'beam:env:' + environment_type.lower()
```

this will allow us to completely get rid of the `Environment.env_type` class attribute.

Reply:
and also we can get rid of `get_urn_from_type()`

Reply:
and `_type_to_urn`